### DISCUSSION TITLE
Prevent ContextLogger from logging bodies that are too large

### DISCUSSION THREAD

**Discussion by CoteArthur** [TypedProject/ts-express-decorators#1717] [Q&A] [answered]
Files are sent to my server in base64 format causing logs that could be multiple MB large when TSED's default filter logs the request (filter as in error handler). How to could I prevent that?

Some solution I thought of :
- Setting a maximum length to logs.
- Completely removing body from ContextLogger (seems unlikely).
- Making a custom filter that catches ALL types of errors to override the default filter and delete/truncate body when logging. Most likely the best one though I couldn't make a fully 'generic' filter (`@Catch(Error)` won't catch Exception / TypeORMError ....).

Thanks !

**Comment by Romakita** [comment, marked as answer]
Hello @CoteArthur 
By default, the body isn't logged (excepted if there is an error). Have you changed the default `logger` configuration?

The Catch decorator isn't responsible to filter the logged body. It's PlatformLogMiddleware the responsible and you can override it to define your own strategy.

See you
Romain

**Reply by CoteArthur** [reply]
Thanks for your answer @Romakita !
I was indeed thinking about logs when an error is thrown.

My idea behind the usage of the Catch decorator was to log the error myself filtering bodies that are too large.
It seems `@Catch(Error)` actually catches TypeORMError so I might be able to make it work with 2 error filters: one for HTTP Exceptions and one for other errors. 

Overriding PlatformLogMiddleware  sounds promising although I couldn't find much ressources on how to do that if you happen to have so lead I'll gladly look into it !

Thanks again

**Reply by Romakita** [reply]
Here is a small example to do that:

```typescript
import {PlatformLogMiddleware, OverrideProvider} from "@tsed/common";

@OverrideProvider(PlatformLogMiddleware)
export CustomPlatformLogMiddleware extends PlatformLogMiddleware {
  /// see PlatformLogMiddleware implementation
}
```
> https://github.com/tsedio/tsed/blob/production/packages/platform/platform-log-middleware/src/middlewares/PlatformLogMiddleware.ts#L107

See you :)

**Reply by CoteArthur** [reply]
Thank you so much ! Worked like a charm !
